pkg/ebpf/events_pipeline.go

```diff
@@ -576,8 +590,13 @@ func (t *Tracee) sinkEvents(ctx context.Context, in <-chan *trace.Event) <-chan
 				continue
 			}
 
+			policies, err := policy.GetSnapshot(event.PoliciesVersion)
```

Comment:
My first attempts were using the policies pointer:

https://github.com/aquasecurity/tracee/pull/3250/files#diff-e74de87a065b53a5a49ea37519b1c19a99f64fa311aef231e728e823a591023cR38

This broke because it has a set of fewer allowed types. To get around this, I replaced unsafe.Pointer with a uint64 value... but this causes panic when the GC releases this pointer. One solution would be to have a dictionary to save this pipeline in use pointer (keeping it hot/referenced) and get it again in the next steps of reading this dictionary. But I assume (not benchmarked) that, in the vast majority of cases, circular buffering will perform better or at worst similar to golang map retrieval (hashing, internal bucket lookup, check return if found, etc). WDYT?

Reply:
Rethinking it, if we keep it hot in a go map, the uit64 field with the right policies version address would be coerced during the whole pipeline instead of re-retrieved.

It would work unlike the event pointer pool which stores unused pointers, this map would store used policy pointers to prevent them from being freed. It might work. LMKWYT.

Reply:
We decided to return to this optimisation in the future.